Basic sidebar setup but still incomplete

MattPereira opened the pull request:
Working on issue #262 recreating [Randy's Sidebar Figma design](https://www.figma.com/file/5C3v1LUNwMPQy9JOgnKEtr/Water-the-Trees?node-id=994%3A2416).

**Issues To Be Solved:**
1. ~~The sidebar (permanent MUI drawer) covers the content of page~~
2. Getting the green arrow button to be positioned exactly how Randy designed in Figma
3. Setting up the "Plant" button to function the same way it does in production currently
4. ~~What to do with "Help" and "Settings" buttons~~ (Removed for now)
5. ~~The icons for "My Trees" and "Plant"~~
6. ~~Displaying user profile pic, name and email when Logged In~~

MattPereira commented:
![Screenshot from 2022-09-29 15-09-23](https://user-images.githubusercontent.com/73561520/193150969-35d2b395-6122-41cc-87fd-55a206f096f0.png)
![Screenshot from 2022-09-29 15-09-03](https://user-images.githubusercontent.com/73561520/193150989-29864fc7-fecc-4b37-896a-dc5c78316e52.png)
![Screenshot from 2022-09-29 15-08-55](https://user-images.githubusercontent.com/73561520/193151006-2559a155-7c26-420f-8f72-dfe8b54f9155.png)

ri0nardo commented:
Regarding the green button to expand and collapse the sidebar, I am not sure if we want to proceed with that design. It is very similar to the green dots to signify a tree. I will look into other options.

The help and settings options are "filler" for the time being. So they won't be included at this phase of the sidebar. We still need to figure out how what to include there.

The icons for "plant" and "my trees", I can export them to the file type that is used for the other icons. Let me know. 
- My trees are only available if you have a profile. So guest won't see that feature.

Right now the profile picture is the "default" initials per user name. Not sure how to merge that into the profile info to the sidebar. See photo attached. 
- Alittle about the profile. includes an email, your name, and profile initals. If you create an account you can find this feature. Only if you have a profile.

For mobile lets go with a hamburger menu. Will want to revisit later when mobile design is the focus.

@MattPereira great job with the sidebar. Looking forward to the little characters. Let me know what format you would want the characters exported.

![12 1](https://user-images.githubusercontent.com/98299952/193438900-f7ac5d9c-15e3-45ee-ab11-4d476758ea54.jpg)

MattPereira commented:
> Good work @MattPereira !! :) Nice use of grid It would be nice to have the drawer be swipeable: https://mui.com/material-ui/react-drawer/ Do we want it to anchor to the top or to a hamburger or tree icon on a small phone? @ri0nardo
> 
> The navigation items should be Links to the route like in the Header.
> 
> Can you make the css into styled components? Personally I prefer stylesheets as its easier to port to a new app and has less clutter but a previous dev moved a ton of stuff to styled components and it is easier to maintain the react app and is easier to see the styles in the component you are working with it. Checkout the tree components for some examples.

I'm a little confused on the swipe-able drawer since Randy confirmed that mobile size will be a hamburger menu? Do you just mean the functionality to swipe up while viewing the hamburger menu?

Also, I converted all of the styles that were in the `Sidebar.scss` file into styled components as you requested

MattPereira commented:
> The help and settings options are "filler" for the time being. So they won't be included at this phase of the sidebar. We still need to figure out how what to include there.
> 
> The icons for "plant" and "my trees", I can export them to the file type that is used for the other icons. Let me know.
> 
> * My trees are only available if you have a profile. So guest won't see that feature.
> 
> Right now the profile picture is the "default" initials per user name. Not sure how to merge that into the profile info to the sidebar. See photo attached.
>
> For mobile lets go with a hamburger menu. Will want to revisit later when mobile design is the focus.

Thanks for sending me the Icons Randy! I added them into the sidebar and updated the screenshots in the comment above.

Also, I removed the "help" and "settings" items from the sidebar for the time being as you suggested.

The "Personal" section of the sidebar now only shows if a user is logged in, and I got the user profile details to show for logged in users

On screen size mobile, the sidebar is hidden and the top bar with hamburger icon shows.

MattPereira commented:
So I finally got the sidebar to stop covering the content of the page but I'm not sure you guys will like how I did it:

There was a `<Box sx={{display:"flex"}}></Box>` wrapping around the `MapLayout.js` component. I moved it to `App.js` to wrap around all of the `<Routes>` and around the `<Sidebar/>` and `<Header />` components.
